distribution_metadata.c

```diff
@@ -594,19 +811,32 @@ TupleToShardPlacement(HeapTuple heapTuple, TupleDesc tupleDescriptor)
 	ShardPlacement *shardPlacement = NULL;
 	bool isNull = false;
 
-	Datum idDatum = heap_getattr(heapTuple, ATTR_NUM_SHARD_PLACEMENT_ID,
+	Datum idDatum = heap_getattr(heapTuple, AttrNumShardPlacementId,
 								 tupleDescriptor, &isNull);
-	Datum shardIdDatum = heap_getattr(heapTuple, ATTR_NUM_SHARD_PLACEMENT_SHARD_ID,
+	Datum shardIdDatum = heap_getattr(heapTuple, AttrNumShardPlacementShardId,
 									  tupleDescriptor, &isNull);
-	Datum shardStateDatum = heap_getattr(heapTuple, ATTR_NUM_SHARD_PLACEMENT_SHARD_STATE,
+	Datum shardStateDatum = heap_getattr(heapTuple, AttrNumShardPlacementShardState,
 										 tupleDescriptor, &isNull);
-	Datum nodeNameDatum = heap_getattr(heapTuple, ATTR_NUM_SHARD_PLACEMENT_NODE_NAME,
+	Datum nodeNameDatum = heap_getattr(heapTuple, AttrNumShardPlacementNodeName,
 									   tupleDescriptor, &isNull);
-	Datum nodePortDatum = heap_getattr(heapTuple, ATTR_NUM_SHARD_PLACEMENT_NODE_PORT,
+	Datum nodePortDatum = heap_getattr(heapTuple, AttrNumShardPlacementNodePort,
 									   tupleDescriptor, &isNull);
 
 	shardPlacement = palloc0(sizeof(ShardPlacement));
-	shardPlacement->id = DatumGetInt64(idDatum);
+
+	if (UseCitusMetadata)
+	{
+		Datum shardLengthDatum = heap_getattr(heapTuple, AttrNumShardPlacementShardLength,
+											  tupleDescriptor, &isNull);
+		shardPlacement->id = (int64) DatumGetObjectId(idDatum);
+		shardPlacement->shardLength = DatumGetInt64(shardLengthDatum);
+	}
+	else
+	{
+		shardPlacement->id = DatumGetInt64(idDatum);
```

Comment:
Yes. Citus does not _have_ placement IDs: it just uses an `Oid`. `Oid` is essentially `int32`, so I have an explicit cast to show that we're using the wider `int64` in the in-memory `struct` representation.